sqlalchemy-stubs/dialects/postgresql/array.pyi

```diff
@@ -0,0 +1,18 @@
+from typing import Any as AnyType, Optional, TypeVar, Type, Callable, overload
+from ...sql import expression
+from ...sql.type_api import TypeEngine
+from ... import types as sqltypes
```

Comment:
Note that all imports of the form `from x import y as z` are treated as public by type checkers. I they are actually "private" (e.g. needed only for annotations), then you should give them a name starting with underscore, for example
```python
from ... import types as _sqltypes
```

Reply:
Is this still the case even if the library uses `from x import y as z`? For instance, [postgres/array.py](https://bitbucket.org/zzzeek/sqlalchemy/src/6e958acf49a606b176e635a216ace95813ffa6a6/lib/sqlalchemy/dialects/postgresql/array.py?at=rel_1_2&fileviewer=file-view-default#array.py-11) import this line as I have it in the typings.

Reply:
Yes, this is still true even if name changes (at least for mypy). The general rule is that names not "officially" advertised as part of public API stay private. For example, `from dataclasses import sys` should be an error even though this line will work at runtime.